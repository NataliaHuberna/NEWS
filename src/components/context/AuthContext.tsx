import React, { useState } from 'react';

type TProps = {
    children: React.ReactNode
};

export const AuthContext = React.createContext(null);

const AuthProvider: React.FC<TProps> = ({ children}) => {
    const [user, setUser] = useState(null);
    const value = {
        user,
        setUser
    };
    
    return <AuthContext.Provider value={value}>
        {children}
    </AuthContext.Provider>;
};

export default AuthProvider;
