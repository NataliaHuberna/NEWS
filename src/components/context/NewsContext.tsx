import React, { useState } from 'react';
import {TNew} from "src/types/newsTypes";

type TProps = {
    children: React.ReactNode
};

export const NewsContext = React.createContext(null);

const NewsProvider: React.FC<TProps> = ({ children}) => {
    const [state, setState] = useState({news:[], singleNew: null});
    const setNews = (news: Array<TNew>) => setState((prevState) => ({...prevState, news}));
    const setSingleNew = (publishedAt: string) => setState((prevState) => ({...prevState,
        singleNew: prevState.news.find((singleNew: TNew) => singleNew.publishedAt === publishedAt )}));
    const value = {
        ...state,
        setNews,
        setSingleNew
    };
    return <NewsContext.Provider value={value}>
        {children}
    </NewsContext.Provider>;

};

export default NewsProvider;
