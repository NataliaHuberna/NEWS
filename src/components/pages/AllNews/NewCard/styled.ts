import styled from "styled-components";

export const StFakeImage = styled.div`
  background: gray;
  height: 140px;
  display: grid; 
  place-items: center;
  color: white;
`;