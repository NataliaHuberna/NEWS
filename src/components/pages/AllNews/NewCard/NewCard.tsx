import * as React from "react";
import CardMedia from "@mui/material/CardMedia";
import {NewsContext} from "src/components/context/NewsContext";
import {StFakeImage} from "src/components/pages/AllNews/styled";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import CardActions from "@mui/material/CardActions";
import {Link} from "react-router-dom";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import {TNew} from "src/types/newsTypes";

type TProps = {
    singleNew: TNew
};

const NewCard: React.FC<TProps> = ({singleNew}) => {
    const { setSingleNew } = React.useContext(NewsContext);
    const {
        urlToImage: urlToImage,
        title: title,
        description: description,
        publishedAt: publishedAt,
    } = singleNew;
    const onClickHandler = () => setSingleNew(publishedAt);

    return (
        <Card sx={{ maxWidth: 345 }} key={publishedAt}>
            {urlToImage ? (
                <CardMedia
                    component="img"
                    height="140"
                    image={urlToImage}
                    alt="green iguana"
                />
            ) : <StFakeImage>Sorry this post dont have image</StFakeImage>}
            <CardContent>
                <Typography gutterBottom variant="h5" component="div">
                    {title}
                </Typography>
                <Typography variant="body2" color="text.secondary">
                    {description}
                </Typography>
            </CardContent>
            <CardActions>
                <Link to={`/news/${publishedAt}`}>
                    <Button size="small" onClick={onClickHandler} >Read More</Button>
                </Link>
            </CardActions>
        </Card>
    );
};

export default NewCard;
