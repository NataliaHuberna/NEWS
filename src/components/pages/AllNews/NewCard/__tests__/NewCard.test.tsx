import React from 'react';
import {mount} from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import NewCard from "../NewCard";
import {NewsContext} from "../../../../context/NewsContext";

type TProps = {
    description: string,
    publishedAt: string,
    title: string,
    urlToImage: string | null
};

type TMockNewsContext = {
    setSingleNew: jest.Mock<any, any>
};

describe('NewCard test', () => {
    const props: TProps = {
        description: 'description',
        publishedAt: 'publishedAt',
        title: 'title',
        urlToImage: 'https://media.cnn.com/api/v1/images/stellar/prod/220216011420-01-missing-girl-found.jpg?c=16x9&q=w_800,c_fill',
    };

    const mockNewsContext: TMockNewsContext = {
        setSingleNew: jest.fn()
    };
    // @ts-ignore
    const sut = (mockValue: TProps, singleNew = props) => (
        mount(
            <BrowserRouter>
                <NewsContext.Provider value={mockValue}>
                    <NewCard singleNew={singleNew}/>
                </NewsContext.Provider>
            </BrowserRouter>
        ));

    it('should render correctly', () => {
        // @ts-ignore
        const wrapper = sut(mockNewsContext);
        expect(wrapper).toMatchSnapshot();
    });

    it('should setSingleNew(publishedAt)', () => {
        // @ts-ignore
        const wrapper = sut(mockNewsContext);
        // @ts-ignore
        wrapper.find('button').props().onClick();
        expect(mockNewsContext.setSingleNew).toHaveBeenCalled();
    });

    it('should render image ', () => {
        // @ts-ignore
        const wrapper = sut(mockNewsContext, {...props, urlToImage: null});
        console.log(wrapper.debug());
        expect(wrapper.find('div div').at(0).text()).toEqual('Sorry this post dont have image');
    });
});
