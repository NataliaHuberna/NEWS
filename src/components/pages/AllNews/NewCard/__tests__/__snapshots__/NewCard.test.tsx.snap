// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCard test should render correctly 1`] = `ReactWrapper {}`;

exports[`NewCard test should setSingleNew p 1`] = `ReactWrapper {}`;

exports[`NewCard test should setSingleNew(publishedAt) 1`] = `ReactWrapper {}`;
